crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -597,7 +598,10 @@ fn get_2q_decomposers_from_target(
                         OperationRef::Standard(_) => (),
                         _ => continue,
                     }
-
+                    // Filter out non-2q-gate candidates
+                    if op.operation.num_qubits() != 2 {
+                        continue;
```

Comment:
it's the same behavior both for 1-qubit and n-qubits gates (n>2)?

Reply:
yes, the gate matrix needs to be 4x4, any n!=2 qubit gate causes the dimension mismatch panic